src/PolarQuad.cpp

```diff
@@ -0,0 +1,631 @@
+
+  /* Initialize memory for arrays */
+  _multiples = new FP_PRECISION[_num_polar];
+
+  /* Compute multiples of sine thetas and weights */
+  for (int p=0; p < _num_polar; p++)
+    _multiples[p] = _sin_thetas[p] * _weights[p];
+}
+
+
+/**
+ * @brief Converts this Quadrature to a character array of its attributes.
+ * @details The character array includes the number of polar angles, the
+ *          the values of the sine and weight of each polar angle, and the 
+ *          product of the sine and weight of each polar angle.
+ * @return a character array of the PolarQuad's attributes
+ */
+std::string PolarQuad::toString() {
+
+  std::stringstream string;
+
+  string << "PolarQuad";
+  string << "\n\t# angles = " << _num_polar;
+
+  string << "\n\tsin thetas = ";
+  if (_sin_thetas != NULL) {
+    for (int p = 0; p < _num_polar; p++)
+      string << _sin_thetas[p] << ", ";
+  }
+
+  string << "\n\tweights = ";
+  if (_weights != NULL) {
+    for (int p = 0; p < _num_polar; p++)
+      string << _weights[p] << ", ";
+  }
+
+  string << "\n\tmultiples= ";
+  if (_multiples != NULL) {
+    for (int p = 0; p < _num_polar; p++)
+      string << _multiples[p] << ", ";
+  }
+
+  return string.str();
+}
+
+
+
+/**
+ * @brief Dummy constructor calls the parent constructor.
+ */
+TYPolarQuad::TYPolarQuad(): PolarQuad() { }
+
+
+/**
+ * @brief Set the number of polar angles to initialize.
+ * @param num_polar the number of polar angles (maximum 3)
+ */
+void TYPolarQuad::setNumPolarAngles(const int num_polar) {
+
+  if (num_polar > 3)
+    log_printf(ERROR, "Unable to set the number of polar angles to %d "
+               "for TYPolarQuad (max 3 angles)", num_polar);
+
+  PolarQuad::setNumPolarAngles(num_polar);
+}
+
+
+/**
+ * @brief Routine to initialize the polar quadrature.
+ * @details This routine uses the tabulated values for the Tabuchi-Yamamoto
+ *          polar angle quadrature, including the sine thetas and weights.
+ */
+void TYPolarQuad::initialize() {
+
+  /* Call parent class initialize routine */
+  PolarQuad::initialize();
+
+  /* Allocate temporary arrays for tabulated quadrature values */
+  FP_PRECISION* sin_thetas = new FP_PRECISION[_num_polar];
+  FP_PRECISION* weights = new FP_PRECISION[_num_polar];
+
+  /* Tabulated values for the sine thetas and weights for the
+   * Tabuchi-Yamamoto polar angle quadrature */
+  if (_num_polar == 1) {
+    sin_thetas[0] = 0.798184;
+    weights[0] = 1.0;
+  }
+
+  else if (_num_polar == 2) {
+    sin_thetas[0] = 0.363900;
+    sin_thetas[1] = 0.899900;
+    weights[0] = 0.212854;
+    weights[1] = 0.787146;
+  }
+
+  else if (_num_polar == 3) {
+    sin_thetas[0] = 0.166648;
+    sin_thetas[1] = 0.537707;
+    sin_thetas[2] = 0.932954;
+    weights[0] = 0.046233;
+    weights[1] = 0.283619;
+    weights[2] = 0.670148;
+  }
+
+  /* Set the arrays of sin thetas and weights */
+  PolarQuad::setSinThetas(sin_thetas, _num_polar);
+  PolarQuad::setWeights(weights, _num_polar);
```

Comment:
Well my intent was to only have allocation/deallocation of the ``_sin_thetas`` and ``_weights`` array in a single place - namely, the ``PolarQuad`` parent class' ``initialize()`` routine. I understand this may seem clunky here since it requires the use of temporary arrays in each subclass' ``initialize()`` routine.